client/app/components/dashboards/AddTextboxDialog.jsx

```diff
@@ -2,154 +2,103 @@ import { markdown } from 'markdown';
-    widget.options.position.row = position.row;
-
-    widget
-      .save()
+    this.props.onConfirm(this.state.text)
       .then(() => {
-        dashboard.widgets.push(widget);
-        this.props.close();
+        this.props.dialog.close();
       })
       .catch(() => {
-        toastr.error('Widget can not be added');
+        toastr.error('Widget could not be added');
       })
       .finally(() => {
         this.setState({ saveInProgress: false });
       });
   }
 
   render() {
-    return (
-      <div>
-        <div className="modal-header">
-          <button
-            type="button"
-            className="close"
-            disabled={this.state.saveInProgress}
-            aria-hidden="true"
-            onClick={this.props.dismiss}
-          >
-            &times;
-          </button>
-          <h4 className="modal-title">Add Textbox</h4>
-        </div>
-        <div className="modal-body">
-          <div className="form-group m-b-0">
-            <textarea
-              className="form-control resize-vertical"
-              style={{ minMeight: '100px' }}
-              rows="5"
-              value={this.state.text}
-              onChange={this.onTextChanged}
-            />
-          </div>
-          <div
-            ng-show="$ctrl.text"
-            className="m-t-15"
-          >
-            <strong>Preview:</strong>
-            <p
-              dangerouslySetInnerHTML={{ __html: this.state.preview }} // eslint-disable-line react/no-danger
-              className="word-wrap-break"
-            />
-          </div>
-        </div>
+    const { dialog } = this.props;
 
-        <div className="modal-footer">
-          <button
-            type="button"
-            className="btn btn-default"
-            disabled={this.state.saveInProgress}
-            onClick={this.props.dismiss}
-          >
-            Close
-          </button>
-          <button
-            type="button"
-            className="btn btn-primary"
-            disabled={this.state.saveInProgress}
-            onClick={() => this.saveWidget()}
-          >
-            Add to Dashboard
-          </button>
+    return (
+      <Modal
+        {...dialog.props}
+        title="Add Textbox"
+        onOk={() => this.saveWidget()}
+        okButtonProps={{
+          loading: this.state.saveInProgress,
+          disabled: !this.state.text,
+        }}
+        okText="Add to Dashboard"
+        width={500}
+      >
+        <div className="add-textbox">
+          <Input.TextArea
+            className="resize-vertical"
+            rows="5"
+            value={this.state.text}
+            onChange={this.onTextChanged}
+            autoFocus
+            placeholder="This is where you write some text"
+          />
+          <small>
+            Supports basic{' '}
+            <a
+              target="_blank"
+              rel="noopener noreferrer"
+              href="https://www.markdownguide.org/cheat-sheet/#basic-syntax"
```

Comment:
Maybe we should move this content to our own page?

Reply:
(not a blocker of course)

Reply:
Hellz yeah, will prepare a help doc and insert a HelpTrigger 💪

Reply:
https://github.com/getredash/website/issues/216